subsys/bootloader/bl_storage/bl_storage.c

```diff
@@ -109,14 +218,7 @@ int public_key_data_read(uint32_t key_idx, uint8_t *p_buf)
 
 	p_key = BL_STORAGE->key_data[key_idx].hash;
 
-	/* Ensure word alignment, since the data is stored in memory region
-	 * with word sized read limitation. Perform both build time and run
-	 * time asserts to catch the issue as soon as possible.
-	 */
-	BUILD_ASSERT(offsetof(struct bl_storage_data, key_data) % 4 == 0);
-	__ASSERT(((uint32_t)p_key % 4 == 0), "Key address is not word aligned");
-
-	otp_copy32(p_buf, (volatile uint32_t *restrict)p_key, SB_PUBLIC_KEY_HASH_LEN);
+	otp_copy(p_buf, p_key, SB_PUBLIC_KEY_HASH_LEN);
```

Comment:
Is it really that we cannot guarantee the word alignment anymore?

Reply:
I restored the check. I thought the check was originally added as the otp_copy32 required the alignment, but now I am unsure about this.